src/__tests__/utils/test-utils.ts

```
@@ -0,0 +1,369 @@
+            )
+        );
+
+        return {
+            authorized: hasPermission,
+            error: hasPermission ? undefined : 'Unauthorized: Insufficient permissions'
+        };
+    });
+
+    jest.mock('@/middleware/check-permission', () => {
+        return {
+            withPermission: (permission: string) => (handler: Function) => async (request: Request) => {
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)